### PR TITLE
Move a few non-critical driver bug tests to 2.0.1

### PR DESCRIPTION
**Pull request by zhenyao**

### CODE REVIEW

**Comment by zhenyao** [comment]
@kenrussell @jdashg @Kangz FYI.  These are all corner case driver issues.  I push them to 2.0.1 or add them in SkipList until driver bugs are fixed (or worked around)

File: sdk/tests/deqp/framework/common/tcuSkipList.js [hunk 1]
```diff
@@ -141,6 +141,30 @@ goog.scope(function() {
         // deqp/functional/gles3/shadertexturefunction/textureprojgradoffset.html
         _skip("texture_functions.textureprojgradoffset.sampler2dshadow_vertex");
         _skip("texture_functions.textureprojgradoffset.sampler2dshadow_fragment");
+
+        _setReason("MacOSX drivers share namespaces where they shoudl not");
```
Comment (Kangz) [inline sdk/tests/deqp/framework/common/tcuSkipList.js]: nit: typo shoudl ->should

File: sdk/tests/deqp/framework/common/tcuSkipList.js [hunk 2]
```diff
@@ -141,6 +141,30 @@ goog.scope(function() {
...
+        // https://github.com/KhronosGroup/WebGL/issues/1890
+        // deqp/data/gles3/shaders/scoping.html
+        _skip("scoping.valid.local_int_variable_hides_struct_type_vertex");
+        _skip("scoping.valid.local_int_variable_hides_struct_type_fragment");
+        _skip("scoping.valid.local_struct_variable_hides_struct_type_vertex");
+        _skip("scoping.valid.local_struct_variable_hides_struct_type_fragment");
+        _skip("scoping.valid.function_parameter_hides_struct_type_vertex");
+        _skip("scoping.valid.function_parameter_hides_struct_type_fragment");
+
+        _setReason("ES spec change to forbid some preprocessor 'defined' uses");
```
Comment (Kangz) [inline sdk/tests/deqp/framework/common/tcuSkipList.js]: We have ANGLE patches in flight or landed that address this, should it still be skipped?

Comment (zhenyao) [inline sdk/tests/deqp/framework/common/tcuSkipList.js]: Once it lands, we can remove the Skip entries.

File: sdk/tests/deqp/framework/common/tcuSkipList.js (hunk 1 above)
Comment (zhenyao) [inline sdk/tests/deqp/framework/common/tcuSkipList.js]: I'll fix this in the next update.

File: sdk/tests/conformance2/glsl3/00_test_list.txt [hunk 3]
```diff
@@ -26,6 +26,6 @@ shader-with-invalid-characters.html
 shader-with-mis-matching-uniform-block.html
 short-circuiting-in-loop-condition.html
 texture-offset-out-of-range.html
-tricky-loop-conditions.html
+--min-version 2.0.1 tricky-loop-conditions.html
 uniform-location-length-limits.html
-vector-dynamic-indexing.html
+--min-version 2.0.1 vector-dynamic-indexing.html
```
Comment (kenrussell) [inline sdk/tests/conformance2/glsl3/00_test_list.txt]: This one is concerning -- guaranteed support for dynamic indexing of vectors and matrices is an important ES 3.0 upgrade. Would it be possible to split off just the failing tests for 2.0.1 (vector-dynamic-indexing-bugs.html?) and enable some of them?

File: sdk/tests/deqp/framework/common/tcuSkipList.js (hunk 2 above)
Comment (kenrussell) [inline sdk/tests/deqp/framework/common/tcuSkipList.js]: Let's make sure not to forget to re-enable them.

**Comment by kenrussell** [comment]
LGTM overall after the fact; couple of comments.

File: sdk/tests/conformance2/glsl3/00_test_list.txt (hunk 3 above)
Comment (zhenyao) [inline sdk/tests/conformance2/glsl3/00_test_list.txt]: Addressed in https://github.com/KhronosGroup/WebGL/pull/1909

File: sdk/tests/deqp/framework/common/tcuSkipList.js (hunk 2 above)
Comment (zhenyao) [inline sdk/tests/deqp/framework/common/tcuSkipList.js]: Ultimately we should clean this SkipList. Let's make it a goal.
